tsl/src/compression/create.c

```diff
@@ -878,16 +880,46 @@ check_modify_compression_options(Hypertable *ht, WithClauseResult *with_clause_o
 			if (fd->segmentby_column_index > 0)
 				segment_by_set = true;
 			if (fd->orderby_column_index > 0)
+			{
 				order_by_set = true;
+				current_orderby_cols = lappend(current_orderby_cols, fd);
+			}
 		}
 		if (with_clause_options[CompressOrderBy].is_default && order_by_set)
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("must specify a column to order by"),
-					 errdetail("The timescaledb.compress_orderby option was"
-							   " previously set and must also be specified"
-							   " in the updated configuration.")));
-
+		{
+			/* If the orderby that's already set is only the time column DESC (which is the
+			 default), and we pass the default again, then no need to give this error */
+			if (list_length(parsed_orderby_cols) != 1)
+				ereport(ERROR,
+						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						 errmsg("must specify a column to order by"),
+						 errdetail("The timescaledb.compress_orderby option was"
+								   " previously set and must also be specified"
+								   " in the updated configuration.")));
+			else
+			{
+				// this is okay only if the orderby that's already set is only the time column
+				// check for the same attribute name and same order (desc)
+				ListCell *elem1 = list_nth_cell(current_orderby_cols, 0);
+				FormData_hypertable_compression *fd_elem1 = lfirst(elem1);
+				NameData colname1 = fd_elem1->attname;
+				ListCell *elem2 = list_nth_cell(parsed_orderby_cols, 0);
+				CompressedParsedCol *cpc_elem2 = lfirst(elem2);
+				NameData colname2 = cpc_elem2->colname;
+				bool orderby_time_default_matches = (fd_elem1->orderby_asc == cpc_elem2->asc);
+
+				if (!(list_length(current_orderby_cols) == 1 &&
+					  list_length(parsed_orderby_cols) == 1 &&
+					  (namestrcmp(&colname1, NameStr(colname2)) == 0) &&
+					  orderby_time_default_matches))
+					ereport(ERROR,
+							(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							 errmsg("must specify a column to order by"),
+							 errdetail("The timescaledb.compress_orderby option was"
+									   " previously set and must also be specified"
+									   " in the updated configuration.")));
+			}
```

Comment:
You can deduplicate this `ereport()` call here.

E.g.
```
if (list_length(parsed_orderby_cols) == 1)
{
  ... do stuff
}

if (!(list_length(current_orderby_cols) == 1 &&
    list_length(parsed_orderby_cols) == 1 &&
    (namestrcmp(&colname1, NameStr(colname2)) == 0) &&
    orderby_time_default_matches))
        ereport(ERROR,
            (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
            errmsg("must specify a column to order by"),
            errdetail("The timescaledb.compress_orderby option was"
                      " previously set and must also be specified"
                      " in the updated configuration.")));
```

Reply:
Thank you. Fixed it